dts/bindings/mipi-dbi/zephyr,mipi-dbi-spi.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  MIPI-DBI Mode C compatible SPI controller. This driver emulates MIPI DBI
+  mode C using a SPI controller and GPIO pins
+compatible: "zephyr,mipi-dbi-spi"
+
+include: ["spi-controller.yaml", "pinctrl-device.yaml"]
+
+properties:
+  spi-dev:
+    type: phandle
+    required: true
+    description: |
+      SPI device to use for data transfers with MIPI DBI commands. This SPI
+      device should be connected to the MIPI DBI display.
+
+  cd-gpios:
+    type: phandle-array
+    description: |
+      Command/data gpio pin. Required when using 4 wire SPI mode (Mode C1).
+      Set to low when sending a command, or high when sending data.
+
+  reset-gpios:
+    type: phandle-array
+    description: |
+      Reset GPIO pin. Used to reset the display during initialization.
+      Active low pin.
```

Comment:
This is too different from the Linux kernel bindings, which Zephyr should follow as closely as possible, https://www.kernel.org/doc/Documentation/devicetree/bindings/display/panel/panel-mipi-dbi-spi.yaml

Reply:
This is a fair point. The issue with the Linux binding (and why we are currently deviating) is that in Linux, the MIPI DBI layer does not seem to allow for custom (IE non-spi) controllers.

Currently, the Linux devicetree for a MIPI-DBI display might look like [this](https://github.com/torvalds/linux/blob/master/arch/riscv/boot/dts/canaan/canaan_kd233.dts#L123):

```
&spi0 {
        pinctrl-0 = <&spi0_pinctrl>;
        pinctrl-names = "default";
        num-cs = <1>;
        cs-gpios = <&gpio0 20 GPIO_ACTIVE_HIGH>;

        panel@0 {
                compatible = "canaan,kd233-tft", "ilitek,ili9341";
                reg = <0>;
                dc-gpios = <&gpio0 21 GPIO_ACTIVE_HIGH>;
                spi-max-frequency = <10000000>;
                status = "disabled";
        };
};
```

The driver for the ILI9341 then calls `mipi_dbi_spi_init` in its probe function: https://github.com/torvalds/linux/blob/master/drivers/gpu/drm/tiny/ili9341.c#L205

This sets up the MIPI DBI device with a backing SPI device, and then functions in the MIPI DBI code can use that to write to the display: https://github.com/torvalds/linux/blob/master/drivers/gpu/drm/drm_mipi_dbi.c#L1103

If the only devices we support are traditional SPI controllers, this approach works fine. The display defines the D/C and RESET GPIOs, and uses the parent SPI controller with `mipi_dbi_spi_init`.

However, this design does  not work if the parent controller isn't a SPI controller, but a pure MIPI DBI device. For example, consider an IP block that supports the following operations:

- Write MIPI DBI command
- Write MIPI DBI data
- Toggle RESET line

The IP automatically manages the RESET and D/C lines, so the display controller should not declare the D/C or RESET pins as GPIO devices. This is why I took the approach of having the Zephyr devicetree look like this:

```
mipi_dbi {
		compatible = "zephyr,mipi-dbi-spi";
		reset-gpios = <&gpio0 6 GPIO_ACTIVE_HIGH>;
		cd-gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
		spi-dev = <&spi2>;
		#address-cells = <1>;
		#size-cells = <0>;

		ili9340: ili9340@0 {
			compatible = "ilitek,ili9340";
			reg = <0>;
			spi-max-frequency = <32000000>;
			rotation = <270>;
			width = <320>;
			height = <240>;
		};
	};
```

With this approach, the display controller has no awareness about the C/D or RESET pin, it expects the MIPI DBI controller to manage them. Moving the C/D and RESET pins to the MIPI controller definition will enable us to support pure MIPI DBI controllers with this framework. Down the line, it would also make supporting 8080 controllers (which also fit into the MIPI DBI API definition nicely) much easier.

Reply:
Okay, but maybe property names can be aligned, s/cd-gpios/dc-gpios? I also suggest to add write-only boolean property.

Reply:
Happy to align property names, I'll make an update for that change